tests/framework-tests/preferences/js/StoreTests.js

```diff
@@ -48,31 +48,56 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
             cookieName: "cookieName",
             payload: "cookieValue",
             cookieOpts: {
+                domain: "example.com",
                 expires: "Fri, 15 Jul 2011 16:44:24 GMT",
-                path: "/"
+                "max-age": 1000,
+                path: "/",
+                samesite: "strict",
+                secure: true
             },
-            expected: "cookieName=cookieValue; expires=Fri, 15 Jul 2011 16:44:24 GMT; path=/"
+            expected: "cookieName=cookieValue; domain=example.com; expires=Fri, 15 Jul 2011 16:44:24 GMT; max-age=1000; path=/; samesite=strict; secure"
         }, {
-            testName: "no expiry date set",
+            testName: "cookie attributes with capitals",
             cookieName: "cookieName",
             payload: "cookieValue",
             cookieOpts: {
-                path: "/"
+                Domain: "example.com",
+                Expires: "Fri, 15 Jul 2011 16:44:24 GMT",
+                "Max-Age": 1000,
+                Path: "/",
+                SameSite: "strict",
+                Secure: true
             },
-            expected: "cookieName=cookieValue; path=/"
+            expected: "cookieName=cookieValue; Domain=example.com; Expires=Fri, 15 Jul 2011 16:44:24 GMT; Max-Age=1000; Path=/; SameSite=strict; Secure"
         }, {
-            testName: "no path set",
+            testName: "secure set to false",
             cookieName: "cookieName",
             payload: "cookieValue",
             cookieOpts: {
-                expires: "Fri, 15 Jul 2011 16:44:24 GMT"
+                secure: false
             },
-            expected: "cookieName=cookieValue; expires=Fri, 15 Jul 2011 16:44:24 GMT"
+            expected: "cookieName=cookieValue"
+        }, {
+            testName: "Secure set to false",
+            cookieName: "cookieName",
+            payload: "cookieValue",
+            cookieOpts: {
+                Secure: false
+            },
+            expected: "cookieName=cookieValue"
         }, {
-            testName: "no path or expiry date set",
+            testName: "only name set",
             cookieName: "cookieName",
             payload: "cookieValue",
             expected: "cookieName=cookieValue"
+        }, {
+            testName: "name attribute in cookie options",
```

Comment:
Probably add a test case that the name attribute in cookie options is applied when `cookieName` is not provided.

Reply:
`fluid.prefs.cookieStore.assembleCookie` doesn't handle the name calculations. That is handled by `fluid.prefs.cookieStore.writeCookie`. There already tests in place against that function. However, I've done a bit of refactoring to make the test reporting a bit clearer, and simplified some other parts of the test code.